tutorials/autotvm/tune_tensor_core_batch_matmul.py

```diff
@@ -0,0 +1,230 @@
+               measure_option=measure_option,
+               callbacks=[autotvm.callback.log_to_file(log_file)])
+
+dispatch_context = autotvm.apply_history_best(log_file)
+best_config = dispatch_context.query(task.target, task.workload)
+print("\nBest config:")
+print(best_config)
+with autotvm.apply_history_best(log_file):
+    with tvm.target.create("cuda"):
+        with tvm.build_config():
+            s, arg_bufs = test_gemm_nn(batch, N, L, M, dtype, layout)
+            print(tvm.lower(s, arg_bufs, simple_mode=True))
+            func = tvm.build(s, arg_bufs)
+dev_module = func.imported_modules[0]
+print(dev_module.get_source())
+
+# check correctness
+if (layout == "NN"):
+  shape_a = (batch, N, L)
+  shape_b = (batch, L, M)
+elif (layout == "NT"):
+  shape_a = (batch, L, N)
+  shape_b = (batch, L, M)
+elif (layout == "TN"):
+  shape_a = (batch, N, L)
+  shape_b = (batch, M, L)
+elif (layout == "TT"):
+  shape_a = (batch, L, N)
+  shape_b = (batch, M, L)
+
+a_np = None
+b_np = None
+c_np = None
+c_np_type = None
+if dtype == 'float16':
+  c_np_type = np.float32
+  a_np = np.random.uniform(size=shape_a).astype(np.float16)
+  b_np = np.random.uniform(size=shape_b).astype(np.float16)
+  c_np = np.zeros((batch, N, M), dtype=np.float32)
+  if (layout == "NN"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :], b_np[bs, :, :])
+  elif (layout == "NT"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].T, b_np[bs, :, :])
+  elif (layout == "TN"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :], b_np[bs, :, :].T)
+  elif (layout == "TT"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].T, b_np[bs, :, :].T)
+elif dtype == 'int8':
+  c_np_type = np.int32
+  a_np = np.random.randint(low=-128, high=127, size=shape_a).astype(np.int8)
+  b_np = np.random.randint(low=-128, high=127, size=shape_b).astype(np.int8)
+  c_np = np.zeros((batch, N, M), dtype=np.int32)
+  if (layout == "NN"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].astype(int32), b_np[bs, :, :].astype(int32))
+  elif (layout == "NT"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].astype(int32).T, b_np[bs, :, :].astype(int32))
+  elif (layout == "TN"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].astype(int32), b_np[bs, :, :].astype(int32).T)
+  elif (layout == "TT"):
+    for bs in range(batch):
+      c_np[bs, :, :] = np.dot(a_np[bs, :, :].astype(int32).T, b_np[bs, :, :].astype(int32).T)
+
+ctx = tvm.gpu()
+c_tvm = tvm.nd.array(np.zeros(c_np.shape, dtype=c_np_type), ctx=ctx)
+a_tvm = tvm.nd.array(a_np, ctx=ctx)
+b_tvm = tvm.nd.array(b_np, ctx=ctx)
+func(a_tvm, b_tvm, c_tvm)
+
+tvm.testing.assert_allclose(c_np, c_tvm.asnumpy(), rtol=1e-3)
+
+evaluator = func.time_evaluator(func.entry_name, ctx, number=100)
+print('Time cost of this operator: %f' % evaluator(a_tvm, b_tvm, c_tvm).mean)
```

Comment:
I have tested the performance on my Titan V GPU, it seems that we can not reach a satisfying performance. In some scenarios, usually large size matmul, we even have the similar speed as non-tensorcore schedule. Perhaps we should add more optimization such as using `storage_align` to reduce bank conflicts.

Reply:
Done. There's really a performance boost, especially on large shapes, after applying storage_align. Thanks!